### DISCUSSION TITLE
/tts/streaming接口并发处理效果

### DISCUSSION THREAD

**Discussion by wutl92** [PaddlePaddle/PaddleSpeech#2211] [Q&A]
现在一次性发起10个请求，短文本20字左右，服务器server处理效果相当于同步出了10个结果，没有异步执行返回base64，有没有办法解决 ？？ 服务器执行多个任务的时候能异步处理？

**Comment by yt605155624** [comment]
对不起 我不明白你的问题，`当于同步出了10个结果` 是什么意思，`服务器执行多个任务的时候能异步处理` 又是什么意思，请描述清楚一些

**Reply by wutl92** [reply]
请求1个文字转语音的时候，只要2S就能返回结果，但是我同时发送10个请求，处理时间就变长了，到最后每个请求的结果同一时间返回的，而且时间比较长。 我想服务器是不是异步处理。有什么办法能让每个请求异步执行转换。10个请求总耗时也只有2-3S 都能返回各自的结果。

**Reply by wutl92** [reply]
也就是感觉没有多线程去处理这个转换。是否可以通过配置文件或者其他方式，让转换多线程处理呢

**Reply by yt605155624** [reply]
开了多线程的，调服务的时候用 htop 看一下 cpu 占用率就知道了，可以把 cpu 的占用情况发一下，目前流式 tts 使用的推理引擎是 onnxruntime-cpu n_threads=4, 理论上如果你的 cpu 核数有 40 多个, 倒是有可能 10 个请求的返回时间和一个请求的返回时间差不多，主要还是看你的机器配置情况
如果有工业级的需求，想要加速或者负载均衡这些的，请使用飞桨部署套件 Paddle Serving 或 FastDeploy 等更专业的工具
